### PR TITLE
Added 2 more verification points in CPU offline and vmbus channel interrupt re-assignement

### PR DESCRIPTION
**Pull request by juhlee-microsoft**
1. Assign the random number to the vmbus' cpu id. The random number is in the range of 0 ~ (n-1). n is the max cpu core counts.
2. Assign the original cpu id to the vmbus channel.
Both do not require the VM reboot.

**TEST RESULTS**

[canonical ubuntuserver 16.04-lts Latest] [LISAv2 Test Results Summary]
[canonical ubuntuserver 16.04-lts Latest] Test Run On           : 05/24/2020 19:40:40
[canonical ubuntuserver 16.04-lts Latest] ARM Image Under Test  : canonical : ubuntuserver : 16.04-lts : Latest
[canonical ubuntuserver 16.04-lts Latest] Initial Kernel Version: 4.15.0-1082-azure
[canonical ubuntuserver 16.04-lts Latest] Final Kernel Version  : 5.6.0+
[canonical ubuntuserver 16.04-lts Latest] Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
[canonical ubuntuserver 16.04-lts Latest] Total Time (dd:hh:mm) : 0:0:22
[canonical ubuntuserver 16.04-lts Latest] 
[canonical ubuntuserver 16.04-lts Latest]    ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
[canonical ubuntuserver 16.04-lts Latest] -------------------------------------------------------------------------------------------------------------------------------------------
[canonical ubuntuserver 16.04-lts Latest]     1 CORE                 CPU-OFFLINE-VMBUS-INTERRUPT-REASSINGMENT                                          PASS                 20.2 

[canonical ubuntuserver 18.04-lts Latest] [LISAv2 Test Results Summary]
[canonical ubuntuserver 18.04-lts Latest] Test Run On           : 05/24/2020 19:40:41
[canonical ubuntuserver 18.04-lts Latest] ARM Image Under Test  : canonical : ubuntuserver : 18.04-lts : Latest
[canonical ubuntuserver 18.04-lts Latest] Initial Kernel Version: 5.3.0-1020-azure
[canonical ubuntuserver 18.04-lts Latest] Final Kernel Version  : 5.6.0+
[canonical ubuntuserver 18.04-lts Latest] Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
[canonical ubuntuserver 18.04-lts Latest] Total Time (dd:hh:mm) : 0:0:26
[canonical ubuntuserver 18.04-lts Latest] 
[canonical ubuntuserver 18.04-lts Latest]    ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
[canonical ubuntuserver 18.04-lts Latest] -------------------------------------------------------------------------------------------------------------------------------------------
[canonical ubuntuserver 18.04-lts Latest]     1 CORE                 CPU-OFFLINE-VMBUS-INTERRUPT-REASSINGMENT                                          PASS                23.03 

[canonical ubuntuserver 16.04-lts Latest] [LISAv2 Test Results Summary]
[canonical ubuntuserver 16.04-lts Latest] Test Run On           : 05/24/2020 20:07:28
[canonical ubuntuserver 16.04-lts Latest] ARM Image Under Test  : canonical : ubuntuserver : 16.04-lts : Latest
[canonical ubuntuserver 16.04-lts Latest] Initial Kernel Version: 4.15.0-1082-azure
[canonical ubuntuserver 16.04-lts Latest] Final Kernel Version  : 5.6.0+
[canonical ubuntuserver 16.04-lts Latest] Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
[canonical ubuntuserver 16.04-lts Latest] Total Time (dd:hh:mm) : 0:1:26
[canonical ubuntuserver 16.04-lts Latest] 
[canonical ubuntuserver 16.04-lts Latest]    ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
[canonical ubuntuserver 16.04-lts Latest] -------------------------------------------------------------------------------------------------------------------------------------------
[canonical ubuntuserver 16.04-lts Latest]     1 CORE                 STRESS-CPU-OFFLINE-ONLINE                                                         PASS                84.21 

[canonical ubuntuserver 18.04-lts Latest] [LISAv2 Test Results Summary]
[canonical ubuntuserver 18.04-lts Latest] Test Run On           : 05/24/2020 20:07:29
[canonical ubuntuserver 18.04-lts Latest] ARM Image Under Test  : canonical : ubuntuserver : 18.04-lts : Latest
[canonical ubuntuserver 18.04-lts Latest] Initial Kernel Version: 5.3.0-1020-azure
[canonical ubuntuserver 18.04-lts Latest] Final Kernel Version  : 5.6.0+
[canonical ubuntuserver 18.04-lts Latest] Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
[canonical ubuntuserver 18.04-lts Latest] Total Time (dd:hh:mm) : 0:1:30
[canonical ubuntuserver 18.04-lts Latest] 
[canonical ubuntuserver 18.04-lts Latest]    ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
[canonical ubuntuserver 18.04-lts Latest] -------------------------------------------------------------------------------------------------------------------------------------------
[canonical ubuntuserver 18.04-lts Latest]     1 CORE                 STRESS-CPU-OFFLINE-ONLINE                                                         PASS                87.24

### CODE REVIEW

File: Testscripts/Linux/channel_change.sh [hunk 1]
```diff
@@ -154,26 +168,113 @@ function Main() {
...
+
+				# Read Rel_ID and target_cpu lines
+				read -a line
+				while [[ $line != '' ]]; do
+					rel_id=${line[0]#"$prefix1"}
+					rel_id=${rel_id%"$suffix1"}
+					LogMsg "Found Rel_ID: $rel_id, which is vmbus channel id"
+					cpu_id=${line[1]#"$prefix2"}
+					LogMsg "Found the original target cpu id: $cpu_id"
+					# This cpu_is is fro lsvmbus.output, the original output.
```
Comment (LiliDeng) [inline Testscripts/Linux/channel_change.sh]: fro for or from?

**Review by LiliDeng — DISMISSED** [review]

File: Testscripts/Linux/channel_change.sh (hunk 1 above)
Comment (juhlee-microsoft) [reply Testscripts/Linux/channel_change.sh]: Fixed in 80240b93a6678b2c1f77ddd6f53163e4410bc0c8

**Comment by juhlee-microsoft** [comment]
**Test Result Update**

[LISAv2 Test Results Summary]
Test Run On           : 05/26/2020 18:58:56
ARM Image Under Test  : canonical : ubuntuserver : 16.04-lts : Latest
Initial Kernel Version: 4.15.0-1082-azure
Final Kernel Version  : 5.6.0+
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:40

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
-------------------------------------------------------------------------------------------------------------------------------------------
    1 CORE                 CPU-OFFLINE-VMBUS-INTERRUPT-REASSINGMENT                                          PASS                37.55 


[LISAv2 Test Results Summary]
Test Run On           : 05/26/2020 18:58:59
ARM Image Under Test  : canonical : ubuntuserver : 18.04-lts : Latest
Initial Kernel Version: 5.3.0-1020-azure
Final Kernel Version  : 5.6.0+
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:27

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
-------------------------------------------------------------------------------------------------------------------------------------------
    1 CORE                 CPU-OFFLINE-VMBUS-INTERRUPT-REASSINGMENT                                          PASS                24.68 


[LISAv2 Test Results Summary]
Test Run On           : 05/26/2020 19:39:22
ARM Image Under Test  : canonical : ubuntuserver : 16.04-lts : Latest
Initial Kernel Version: 4.15.0-1082-azure
Final Kernel Version  : 5.6.0+
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:1:50

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
-------------------------------------------------------------------------------------------------------------------------------------------
    1 CORE                 STRESS-CPU-OFFLINE-ONLINE                                                         PASS               108.05 


[LISAv2 Test Results Summary]
Test Run On           : 05/26/2020 19:39:25
ARM Image Under Test  : canonical : ubuntuserver : 18.04-lts : Latest
Initial Kernel Version: 5.3.0-1020-azure
Final Kernel Version  : 5.6.0+
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:1:45

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
-------------------------------------------------------------------------------------------------------------------------------------------
    1 CORE                 STRESS-CPU-OFFLINE-ONLINE                                                         PASS               102.01

File: Testscripts/Linux/channel_change.sh [hunk 2]
```diff
@@ -135,59 +143,137 @@ function Main() {
...
+			else
+				LogErr "Failed. Supposed to read Sysfs path, but found ${line[@]}"
+				FailedCount=$((FailedCount+1))
+			fi
+			idx=$((idx+1))
+		else
+			LogErr "Failed. Supposed to read VMBUS ID line, but found ${line[@]}"
+			FailedCount=$((FailedCount+1))
+		fi
+		LogMsg ""
```
Comment (abhimarathe) [inline Testscripts/Linux/channel_change.sh]: Why do we need a blank log?

Comment (juhlee-microsoft) [reply Testscripts/Linux/channel_change.sh]: There are many vmbus channels in VM. The number of vmbus channels and the number of cpu id are dependent on the VM size. Looping many times for all cpu id, it is good to have blank line for easy read. If you don't like it, I can remove it.

Comment (abhimarathe) [reply Testscripts/Linux/channel_change.sh]: Can we have a log message instead of blank line? like Verification complete?

**Review by abhimarathe — DISMISSED** [review]

**Review by abhimarathe — APPROVED** [review]

**Pull request merged**